### ISSUE TITLE
Improvements to the release process/CI

### ISSUE THREAD

**Issue by denschub** [denschub/camo-rs#15]
There are a couple of things I could improve.

- [x] The binary inside the archive doesn't have the executable flag set. This is [due to an issue in the `upload-artifact` action](https://github.com/actions/upload-artifact/issues/38), but this can be worked around by building a `tar`/`zip` manually and uploading that.
- [x] Instead of `camo-linux-amd64.zip`, it probably makes sense to add the version number to the archive, something like `camo-v0.3.0-linux-amd64.zip` so it's clear when linking to it.
- [ ] For releasing, I currently have to wait until the `build` job for the `release` branch is done, then downloading the artifacts, drafting a release, and attaching the artifacts there. In theory, I could run GHA [when a new tag is created](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#create), build release artifacts, and automatically create a new draft release with the artifacts attached. The `matrix-org/synapse` repo [has something close to that](https://github.com/matrix-org/synapse/blob/fa2f3d8d0c915c6ebd84ef53526ac04c91fd4a19/.github/workflows/release-artifacts.yml#L155) where I could probably take inspiration from.
- [x] My current version number hacks based on `::set-output` [are deprecated, and there is a much nicer way available](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).
- [x] `action-rs` seems unmaintained. This will become [an issue soon since GitHub Actions will deprecate NodeJS 12](https://github.com/actions-rs/toolchain/issues/219). I can probably just manually use `rustup` and call `cargo`...

Optionally, some other things:

- [ ] The `docker-release` workflow currently doesn't care about the "pre-release" checkbox in a release, and would tag even pre-releases to `latest`. This probably isn't a big deal since I don't plan on releasing pre-releases, but it might be worth adjusting that, and only push a tag `1.0.0-dev`, instead of pushing it to `1`, `1.0`, `1.0.0-dev`, and `latest`.
- [x] I currently push container images to Docker Hub, but there's no reason not to push it towards the GitHub Container Registry as well.

**Comment by denschub** [comment]
I've made some of the required changes (see the checklist in the initial issue), but I'll postpone some others, mainly the "maintainer comfort" things. GitHub Actions isn't my favorite CI, and I feel like to comfortably do this without too many hacks, I'd need to invent my own action(s). This isn't needed for a 1.0.0 release, so I'll unblock the release.

**Comment by denschub** [comment]
I'm currently too lazy to address any of the remaining issues, mainly because I don't have a need for now.
